src/impl_constructors.rs

```diff
@@ -171,6 +171,19 @@ where
         }
         eye
     }
+
+    /// Create a 2D matrix from its diagonal
```

Comment:
We should add a note about the panic conditions, and it would be nice to have an example:

```rust
/// Create a 2D matrix from its diagonal.
///
/// **Panics** if `diag.len() * diag.len()` would overflow `isize`.
///
/// ```rust
/// // example here
/// ```
```